reaktive/src/commonMain/kotlin/com/badoo/reaktive/disposable/DisposableScope.kt

```diff
@@ -0,0 +1,43 @@
+package com.badoo.reaktive.disposable
+
+import com.badoo.reaktive.completable.Completable
+import com.badoo.reaktive.maybe.Maybe
+import com.badoo.reaktive.observable.Observable
+import com.badoo.reaktive.single.Single
+
+interface DisposableScope : Disposable {
+
+    fun <T : Disposable> T.scope(): T
+
+    fun cancel()
+
+    fun <T> Observable<T>.subscribeScoped(
```

Comment:
I don't think we need them. What is the problem with `scope()`?

Reply:
Why do you think so? I find them very handy.

Reply:
The problems with `subscribeScoped` are: they override current approach of using `subscribe` – it is much easier to try use them when you don't need or can't, and forget to use them when needed. `scope()` is more explicit in this way, especially in combination with `@CheckResult`. Also you need to support both subscribe variants.

Reply:
Well, If you are inside a scope then I hardly imagine that you will need just `subscribe` without cancellation (very rare case).
And if you forget to use `subscribeScoped` then you could forget to use `scope()` as well - but in both cases `@CheckResult` would help.

Reply:
I just very like one with automatic disposable management

Reply:
Yep, looks convenient in my opinion.